sqldelight-compiler/src/test/kotlin/com/squareup/sqldelight/core/QueriesTypeTest.kt

```diff
@@ -132,4 +134,126 @@ class QueriesTypeTest {
       |}
       |""".trimMargin())
   }
+
+    @Test fun `queries file is generated properly via compilation1a`() {
+        val result = FixtureCompiler.compileSql("""
+      |CREATE VIRTUAL TABLE data USING fts5(
+      |  id INTEGER PRIMARY KEY,
+      |  value TEXT AS kotlin.collections.List
+      |);
+      |
+      |insertData:
+      |INSERT INTO data
+      |VALUES (?, ?);
+      |
+      |selectForId:
+      |SELECT *
+      |FROM data
+      |WHERE id = ?;
+    """.trimMargin(), temporaryFolder, fileName = "Data.sq")
+
+        val select = result.compiledFile.namedQueries.first()
+        val insert = result.compiledFile.namedMutators.first()
+        assertThat(result.errors).isEmpty()
+
+        val dataQueries = File(result.outputDirectory, "com/example/testmodule/TestDatabaseImpl.kt")
+        assertThat(result.compilerOutput).containsKey(dataQueries)
+        assertThat(result.compilerOutput[dataQueries].toString()).isEqualTo("""
+      |package com.example.testmodule
+      |
+      |import com.example.Data
+      |import com.example.DataQueries
+      |import com.example.TestDatabase
+      |import com.squareup.sqldelight.Query
+      |import com.squareup.sqldelight.TransacterImpl
+      |import com.squareup.sqldelight.db.SqlCursor
+      |import com.squareup.sqldelight.db.SqlDriver
+      |import com.squareup.sqldelight.internal.copyOnWriteList
+      |import kotlin.Any
+      |import kotlin.Int
+      |import kotlin.Long
+      |import kotlin.String
+      |import kotlin.collections.List
+      |import kotlin.collections.MutableList
+      |import kotlin.jvm.JvmField
+      |import kotlin.reflect.KClass
+      |
+      |internal val KClass<TestDatabase>.schema: SqlDriver.Schema
+      |  get() = TestDatabaseImpl.Schema
+      |
+      |internal fun KClass<TestDatabase>.newInstance(driver: SqlDriver): TestDatabase =
+      |    TestDatabaseImpl(driver)
+      |
+      |private class TestDatabaseImpl(
+      |  driver: SqlDriver
```

Comment:
Something else is still broken, there should be dataAdapter in this constructor

Reply:
Yea, I did think it to be a bit strange when I pulled that part of the code out to make the test pass.[ I did that at the same time](https://github.com/cashapp/sqldelight/pull/1790/commits/9fe4576ca587f16819fd180a15e60c4cdc499478#diff-934599e5af7d79162a1e4676c84e649bL189) that I removed the types off of the generated sql.

I'll take a look to see if I can find the method you allude to in your other comment.